.github/libs/promiseWhile.ts

```diff
@@ -30,6 +30,11 @@ function promiseDoWhile(condition: () => boolean, action: (() => Promise<void>)
         console.info('[promiseWhile] promiseDoWhile() condition', condition);
         const actionResult = action?.();
         console.info('[promiseWhile] promiseDoWhile() actionResult', actionResult);
+
+        if (actionResult === undefined) {
+            resolve();
```

Comment:
Wow, great find @JKobrynski. I'm only curious if we should `resolve` here or `reject` here?

Reply:
Great question, I'm going to ask for some opinions

Reply:
A slight adjustment:
```ts
function promiseDoWhile(condition: () => boolean, action: (() => Promise<void>) | DebouncedFunc<() => Promise<void>> | undefined): Promise<void> {
    console.info('[promiseWhile] promiseDoWhile()');

    return new Promise((resolve, reject) => {
        console.info('[promiseWhile] promiseDoWhile() condition', condition);
        const actionResult = action?.();
        console.info('[promiseWhile] promiseDoWhile() actionResult', actionResult);
        if (!actionResult) {
            resolve();
            return;
        }

        actionResult
            .then(() => promiseWhile(condition, action))
            .then(resolve)
            .catch(reject);
    });
}
```